src/nimblepkg/version.nim

```diff
@@ -141,8 +144,8 @@ proc withinRange*(ver: Version, ran: VersionRange): bool =
     return ver <= ran.ver
   of verEq:
     return ver == ran.ver
-  of verSpecial:
-    return ver == ran.spe
+  of verSpecial: # special version is always within range
+    return ver.isSpecial and ver == ran.spe or not ver.isSpecial
```

Comment:
btw is this the tag name or the resolved hash?

Reply:
afaict its whatever follows `#`i.e. `#head` `#branch` `#commit`

Reply:
I think the ideal here is that the actual resolved version is used - that exploits de-facto compatibility between two branches, should they happen to point to the same commit

Reply:
`withinRange` is too fundamental, it needs to be fast and be ready to do the lookups quite early on. But we can explore the idea at some point I guess. Right now there is a huge ordering issue that needs to be addressed with a major refactor